mmdet/core/evaluation/eval_hooks.py

```diff
@@ -12,28 +15,76 @@ class EvalHook(Hook):
         interval (int): Evaluation interval (by epochs). Default: 1.
     """
 
-    def __init__(self, dataloader, interval=1, **eval_kwargs):
+    def __init__(
+        self,
+        dataloader,
+        interval=1,
+        save_optimizer=True,
+        checkpoint_metrics=["bbox_mAP", "segm_mAP"],
+        out_dir=None,
+        **eval_kwargs
+    ):
         if not isinstance(dataloader, DataLoader):
             raise TypeError(
-                'dataloader must be a pytorch DataLoader, but got {}'.format(
-                    type(dataloader)))
+                "dataloader must be a pytorch DataLoader, but got {}".format(
+                    type(dataloader)
+                )
+            )
         self.dataloader = dataloader
         self.interval = interval
+        self.save_optimizer = save_optimizer
+        # Best model metrics (value, epoch)
+        self.best_model_metrics = defaultdict(lambda: (0, 0))
+        self.checkpoint_metrics = checkpoint_metrics
+        self.out_dir = out_dir
+        # best_metric_value_epoch
+        self.filename_tmpl = "best_{}_{}_{{}}"
```

Comment:
Only two elements are used in self.checkpoint. So, maybe the last one is unnecessary.

Reply:
The format is used later in `runner.save_checkpoint`, see `filename_tmpl = 'epoch {}`